barman/backup.py

```diff
@@ -1628,3 +1634,62 @@ def verify_backup(self, backup_info):
             output.error(e.args[0]["err"])
             return
         output.info(pg_verifybackup.get_output()[0].strip())
+
+    def combine_backups(self, backup_info, backup_name=None, dest_dir=None):
+        """
+        Combines a backup chain into a single synthetic backup using the
+        pg_combinebackup utility.
+
+        :param barman.infofile.LocalBackupInfo backup_info: the last backup
+            in the chain.
+        :param str dest_dir: the directory where the synthetic backup is going
+            to me mounted on. If not passed, the server backup directory is used
+            and the synthetic backup will be a new backup in the backup catalog.
+        """
+        pass
+
+    def create_backup_info(self, backup_info, synthetic_backup_dir):
+        """
+        Generates a backup_info file for a synthetic backup.
+
+        :param barman.infofile.LocalBackupInfo backup_info: the last incremental
+            backup of the chain used to generate the synthetic backup.
+        :param str synthetic_backup_dir: the synthetic backup directory
+        """
+        synthetic_backup_info = SynthethicBackupInfo(
+            server=self.server,
+            backup_dir=synthetic_backup_dir,
+        )
+
+        # set server parameters
+        strategy = PostgresCombineBackupStrategy(self.server.postgres, self.config.name)
+        strategy.set_pg_metadata(synthetic_backup_info)
+
+        # set wal-related parameters from the last incremetal backup
+        wal_fields = (
+            "begin_wal",
+            "begin_xlog",
+            "begin_offset",
+            "end_wal",
+            "end_xlog",
+            "end_offset",
+        )
+        for field in wal_fields:
+            last_backup_val = getattr(backup_info, field, None)
+            synthetic_backup_info.set_attribute(field, last_backup_val)
+
+        # calculate the backup size
+        backup_size = 0
+        backup_dest = synthetic_backup_info.get_basebackup_directory()
+        for dir_path, _, file_names in os.walk(backup_dest):
```

Comment:
Maybe this can be done using the walk_to_root method and summing the size field from backup.info?! This is looping over the dir and opening each file to get the stat. Dunno, just a thought!

Reply:
Hmmmm I'm not sure about it, but I don't think the `pg_combinebackup` works as simple as putting all data together, I wouldn't assume the synthetic backup size will always be exactly the sum of all backups in the chain 🤔